### PR TITLE
Reapply 8644 on 9260

### PR DESCRIPTION
**Pull request by aakselrod**
Rebase of #9242 on #9260. Now includes btcsuite/btcwallet#967.

### CODE REVIEW

**Comment by aakselrod** [comment]
Looks like a postgres itest failed. Looking into it. Doesn't appear to be related to the btcwallet deadlock fixed recently, so that's good!

**Comment by aakselrod** [comment]
It looks like the shutdown came too early and the sweeper in `monitorFeeBumpResult` didn't get time to process the result of the sweep broadcast (exited at `sweeper.go:1665`). I don't think that's related to the SQL DB?

**Comment by aakselrod** [comment]
Updated to address comments, add release notes, and do a CI run. I'm still running this locally to see if the new DB settings give me any trouble.

**Comment by aakselrod** [comment]
Got a postgres unit test failure. Looking into it...

**Comment by yyforyongyu** [comment]
> I believe this is actually related to a flake in chainntnfs where a certain series of events makes the notifier attempt to send one more event than the (unread) channel has room for. 

Thanks a lot for digging into the unit test flake! That certainly was an easy way to fix and was originally used in #9258. Then I realized there's another place we need the dedup check which was fixed in https://github.com/lightningnetwork/lnd/pull/9277/commits/463204408cbfd45c2c7fc51d8329ca9bc739eea4. Since this PR is based on #9260 and that PR has this commit in its upstream, I'd expect it to be fixed properly there but it seems not? The other related fix is #9309 which fixed another flake.

Moving forward, I think we can focus on getting the postgres-related merged, then open new PRs to fix the unit test flakes.

File: Makefile [hunk 1]
```diff
@@ -195,9 +195,11 @@ ifeq ($(dbbackend),postgres)
...
 
 	# Start a fresh postgres instance. Allow a maximum of 500 connections so
-	# that multiple lnd instances with a maximum number of connections of 50
-	# each can run concurrently.
-	docker run --name lnd-postgres -e POSTGRES_PASSWORD=postgres -p 6432:5432 -d postgres:13-alpine -N 500
+	# that multiple lnd instances with a maximum number of connections of 20
+	# each can run concurrently. Note that many of the settings here are
+	# specifically for integration testing and are not fit for running
+	# production nodes.
+	docker run --name lnd-postgres -e POSTGRES_PASSWORD=postgres -p 6432:5432 -d postgres:13-alpine -N 1500 -c max_pred_locks_per_transaction=1024 -c max_locks_per_transaction=128 -c jit=off -c work_mem=8MB -c checkpoint_timeout=10min -c enable_seqscan=off
```
[the remark this answers is not in the document]
Comment (djkazic) [reply Makefile]: Yeah, for context those are the settings I'm using in my production postgres backed LND. DB size is ~5GB.

**Comment by aakselrod** [comment]
I tried to take off the latest commit and force-push but GitHub got stuck on `processing latest update` so I rearranged the order of the remaining commits a little and force-pushed again.

**Comment by aakselrod** [comment]
The only test that failed looks irrelevant to the DB, woohoo!

**Comment by yyforyongyu** [comment]
Could you do a rebase on `yy-beat-itest-optimize`? I made some updates there (mostly just rebase on master), and, it now almost always fails at the postgres-related itests. I also port this PR on top of it [here](https://github.com/yyforyongyu/lnd/pull/19), and the postgres-related itests are now passing. Think we just need to do a rebase and show other reviewers that this PR fixes the related itests.

**Comment by aakselrod** [comment]
Yep, I'll rebase this one, and then backport #9242/rebase it on current master. Thanks!

**Comment by aakselrod** [comment]
I've rebased this PR. Once CI runs, I'll backport/rebase #9242 to get it ready for merge. Thanks!

**Comment by aakselrod** [comment]
The cross compilation test failed due to lack of disk space, which is also happening on #9260 CI runs. Looking into the postgres itest that failed.

**Comment by aakselrod** [comment]
In the itest that failed, it looks like maybe the funding manager is deadlocked. Going to run flakehunter locally to see if I can reproduce and get a goroutine dump from the node that deadlocked (Dave in this case, I think).

**Comment by yyforyongyu** [comment]
Also breifly looked at the logs, think the below 90s is weird,

From Bob's log,
```
2024-12-05 23:47:37.243 [DBG] RPCS interceptor.go:775: [/lnrpc.Lightning/GetInfo] requested
2024-12-05 23:47:55.906 [DBG] HLCK healthcheck.go:330: Health check: chain backend invoking success callback
2024-12-05 23:48:36.841 [INF] GRPH builder.go:915: Processed channels=3 updates=5 nodes=0 in last 1m40.890821807s
2024-12-05 23:48:55.906 [DBG] HLCK healthcheck.go:330: Health check: chain backend invoking success callback
```

From Carol's log,
```
2024-12-05 23:47:37.481 [DBG] RPCS interceptor.go:775: [/lnrpc.Lightning/GetInfo] requested
2024-12-05 23:48:04.463 [DBG] HLCK healthcheck.go:330: Health check: chain backend invoking success callback
2024-12-05 23:48:36.865 [INF] GRPH builder.go:915: Processed channels=3 updates=6 nodes=0 in last 1m32.368310393s
2024-12-05 23:49:04.462 [DBG] HLCK healthcheck.go:330: Health check: chain backend invoking success callback
```

From Eve's log,
```
2024-12-05 23:47:37.715 [DBG] RPCS interceptor.go:775: [/lnrpc.Lightning/GetInfo] requested
2024-12-05 23:48:07.173 [ERR] NANN chan_status_manager.go:519: Unable to retrieve chan status for Channel(096341c34446f64bcac7215be98c78ab5a43f0ab3858c8da102e9681808b31d6:0): edge not found: op=096341c34446f64bcac7215be98c78ab5a43f0ab3858c8da102e9681808b31d6:0
2024-12-05 23:48:11.212 [DBG] HLCK healthcheck.go:330: Health check: chain backend invoking success callback
2024-12-05 23:49:07.176 [ERR] NANN chan_status_manager.go:519: Unable to retrieve chan status for Channel(096341c34446f64bcac7215be98c78ab5a43f0ab3858c8da102e9681808b31d6:0): edge not found: op=096341c34446f64bcac7215be98c78ab5a43f0ab3858c8da102e9681808b31d6:0
```

Looks like they are all blocked by `GetInfo`.

**Comment by aakselrod** [comment]
These are really similar symptoms to when we had the `btcwallet` deadlock. I've added some logging but had no luck reproducing the failure locally when running `flakehunter`. I'm going to rebase on your latest branch update, and push a commit that logs a goroutine dump when `funding.(*Manager)` gets shut down, and then hopefully we get a failure in CI.

**Comment by aakselrod** [comment]
No postgres itest errors this run, only linter complaints about line length in the temporary debug commit. I'm fixing the lint errors and pushing again to try to get an itest failure.

**Comment by aakselrod** [comment]
I think the `unit-race` error is OK since it's just a 10-second timeout, but I was a little suspicious because there are sqlite and postgres connections involved. From an initial look, though, seems like it was just too slow. I'll avoid force-pushing the next temporary commit to keep this log visible, but I'll make a minor change just to get a CI run again.

**Comment by aakselrod** [comment]
This time, the error looks to be due to the fact that the payment attempts were sent while the channel between Carol and Dave was still re-establishing, causing an insufficient balance/no route error because there was not yet an active channel between the nodes.

**Comment by aakselrod** [comment]
Added handling for `commit unexpectedly resulted in rollback` (`pgx.ErrTxCommitRollback) as a serialization error and rebased on latest #9260.

**Comment by aakselrod** [comment]
Zipping postgres log file on failure didn't work in the failed postgres itest, because it's not in a separate matrix-based workflow from the basic itests. I didn't notice this when rebasing. Will fixup and force push here and in #9242.

**Comment by aakselrod** [comment]
Rebased and added logging for `kvdb/sqlbase`.

**Comment by yyforyongyu** [comment]
hmm think it's closed once the upstream branch was deleted...since we are merging #9242 so think it's fine to close it.

**Pull request closed without merging**